pkg/registry/core/pod/strategy.go

```diff
@@ -281,6 +272,58 @@ func (podEphemeralContainersStrategy) WarningsOnUpdate(ctx context.Context, obj,
 	return nil
 }
 
+type podResizeStrategy struct {
+	podStrategy
+}
+
+// ResizeStrategy wraps and exports the used podStrategy for the storage package.
+var ResizeStrategy = podResizeStrategy{Strategy}
+
+// dropNonPodResizeUpdates discards all changes except for pod.Spec.Containers[*].Resources,ResizePolicy and certain metadata
+func dropNonPodResizeUpdates(newPod, oldPod *api.Pod) *api.Pod {
+	pod := oldPod.DeepCopy()
+	pod.Name = newPod.Name
+	pod.Namespace = newPod.Namespace
+	pod.ResourceVersion = newPod.ResourceVersion
+	pod.UID = newPod.UID
+
+	oldCtrToIndex := make(map[string]int)
+	for idx, ctr := range pod.Spec.Containers {
+		oldCtrToIndex[ctr.Name] = idx
+	}
+	for _, ctr := range newPod.Spec.Containers {
+		idx, ok := oldCtrToIndex[ctr.Name]
+		if !ok {
+			continue
+		}
+		pod.Spec.Containers[idx].Resources = ctr.Resources
+		pod.Spec.Containers[idx].ResizePolicy = ctr.ResizePolicy
+	}
```

Comment:
I think doing this in `PrepareForUpdate` means any other changes will get silently dropped. That's probably OK for status & metadata, but for spec it might be better to fail in validation.

Reply:
Not only for `status` and `metadata`, we have the same behavior for `ephemeralcontainers` subresource. Hence, I decided to silently drop. WDYT?

Reply:
I was also wondering the approach here.  I saw `"pod resize may not change fields other than cpu and memory"` in validation but also saw `dropNonPodResizeUpdates` and wasn't clear which approach was being taken here.

FWIW, we've always wiped status when updating spec and vis-versa, and the `ephemeralcontainers` precedence is compelling.  I think I'm in favor of wiping non-updated fields, purely for consistency reasons.  But curious what others think.

Reply:
The part I'm confused about is:  Are changes to "fields other than cpu and memory" being dropped and ignored, or are such changes being reported as errors?  Or both?

Reply:
If they are dropped, then we can kind of pretend they don't exist, and users can specify ONLY the changed fields, right?  That seems better than validating it and risking missing things.  Sort of treat it as unknown fields?